aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -838,50 +880,45 @@ Tensor outer(const Tensor& self, const Tensor& vec2) {
   return self.reshape({self.size(0), 1}) * vec2;
 }
 
-static void addmm_impl_cpu_(
-    Tensor &result, const Tensor &self, Tensor m1, Tensor m2, const Scalar& beta, const Scalar& alpha) {
-  TORCH_INTERNAL_ASSERT(self.dim() == 2 && m1.dim() == 2 && m2.dim() == 2);
+TORCH_IMPL_FUNC(addmm_cpu_out)(const Tensor& self, const Tensor& m1,
+                const Tensor& m2, const Scalar& beta, const Scalar& alpha,
+                const Tensor &result) {
+  c10::MaybeOwned<Tensor> self_ = expand_size(self, {m1.sizes()[0], m2.sizes()[1]});
+  auto betaval = beta.toComplexDouble();
 
   // Array access is faster than .size(n) and .stride(n)
   const auto self_sizes = self.sizes();
   auto m1_strides = m1.strides();
   auto m1_sizes = m1.sizes();
   auto m2_strides = m2.strides();
   auto m2_sizes = m2.sizes();
-
-  TORCH_CHECK(
-      m1_sizes[1] == m2_sizes[0], "mat1 and mat2 shapes cannot be multiplied (",
-      m1_sizes[0], "x", m1_sizes[1], " and ", m2_sizes[0], "x", m2_sizes[1], ")");
-
-  TORCH_CHECK(
-      self_sizes[0] == m1_sizes[0] && self_sizes[1] == m2_sizes[1],
-      "input shape is incompatible with matrix multiplication (",
-      m1_sizes[0], "x", m1_sizes[1], " @ ", m2_sizes[0], "x", m2_sizes[1], " != ",
-      self_sizes[0], "x", self_sizes[1], ")");
-
-  at::native::resize_output(result, self_sizes);
   const auto result_strides = result.strides();
   const auto result_sizes = result.sizes();
 
   if (result.numel() == 0) {
     return;
   }
 
-  if (beta.toComplexDouble() != 0.0 && !self.is_same(result)) {
-    result.copy_(self);
+  if (&result != &self) {
+    self_ = expand_size(self, {m1_sizes[0], m2_sizes[1]}, "addmm");
+  } else {
+    self_ = c10::MaybeOwned<Tensor>::borrowed(self);
```

Comment:
TODO: Fix this bug - inplace variant isn't resized if number of elements are zero.